snippets/javascript.json

```diff
@@ -653,5 +653,598 @@
+    ],
+    "description": "Add an OGCFeatureLayer with URL and collection ID. OGCFeatureLayer only supports collection items encoded in GeoJSON. AMD path: esri/layers/OGCFeatureLayer | ESM path: @arcgis/core/layers/OGCFeatureLayer.js",
+    "prefix": "OGCFeatureLayer"
+  },
+  "Add RouteLayer": {
+      "body": [
+        "new RouteLayer({",
+    "\tstops: [",
+    "\t\t{ geometry: { x: ${1:-117.1825}, y: ${2:34.054722} } },",
+    "\t\t{ geometry: { x: ${3:-116.545278}, y: ${4:33.830278} } }",
+    "\t],",
+    "\tpointBarriers: [",
+    "\t\t{ geometry: { x: ${5:-117.2}, y: ${6:34.05} } }",
+    "\t],",
+    "\tpolylineBarriers: [",
+    "\t\t{ geometry: { paths: [[[${7:-117.3}, ${8:34.05}], [${9:-117.5}, ${10:34.1}]]] } }",
+    "\t],",
+    "\tpolygonBarriers: [",
+    "\t\t{ geometry: { rings: [[[${11:-117.3}, ${12:34.05}], [${13:-117.5}, ${14:34.1}], [${15:-117.4}, ${16:34.2}], [${11}, ${12}]]] } }",
+    "\t]",
+    "});"
+      ],
+      "description": "Add a RouteLayer and visualize the route using stops, point, polygon and polyline barriers. The results of a solved route include overall travel time, distance, and turn-by-turn directions. AMD path: esri/layers/RouteLayer | ESM path: @arcgis/core/layers/RouteLayer.js",
+      "prefix": "routeLayer"
+    },
+  "Add Stream Layer with a URL": {
+    "body": [
+    "new StreamLayer({",
+    "\turl: \"${1:https://geoeventsample1.esri.com:6443/arcgis/rest/services/LABus/StreamServer}\",",
+    "\tpurgeOptions: {",
+    "\t\tdisplayCount: ${2:10000}",
+    "\t},",
+    "\tmaxReconnectionAttempts: ${3:100},",
+    "\tmaxReconnectionInterval: ${4:10},",
+    "\trenderer: ${5:renderer}",
+    "});"
+    ],
+    "description": "Add a StreamLayer with a URL. A StreamLayer displays the observation streams, including changes to location, attributes, or both, associated with a set of tracked objects in real-time. AMD path: esri/layers/StreamLayer | ESM path: @arcgis/core/layers/StreamLayer.js",
+    "prefix": "StreamLayer"
+  },
+  "Add Stream Layer with WebSocketURL": {
+    "body": [
+    "new StreamLayer({",
+    "\twebSocketUrl: \"${3:ws://localhost:8000}\",",
+    "\tfields: [",
+    "\t\t{",
+    "\t\t\tname: \"OBJECTID\",",
+    "\t\t\talias: \"ObjectId\",",
+    "\t\t\ttype: \"oid\"",
+    "\t\t},",
+    "\t\t{",
+    "\t\t\tname: \"TRACKID\",",
+    "\t\t\talias: \"TrackId\",",
+    "\t\t\ttype: \"long\"",
+    "\t\t}",
+    "\t],",
+    "\ttimeInfo: {",
+    "\t\ttrackIdField: \"TRACKID\"",
+    "\t},",
+    "\tgeometryType: \"point\",",
+    "\tmaxReconnectionAttempts: ${3:100},",
+    "\tmaxReconnectionInterval: ${4:10},",
+    "\trenderer: ${5:renderer}",
+    "});"
+    ],
+    "description": "Add a StreamLayer by referencing a custom stream service. The geometryType property is used to set the features' geometry type. AMD path: esri/layers/StreamLayer | ESM path: @arcgis/core/layers/StreamLayer.js",
+    "prefix": "StreamLayerwithWebSocketURL"
+  },
+  "Add Vector Tile Layer": {
```

Comment:
Constructing the vector tile using the style object is OK, but pretty unusual.  I would add [the usual one](https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-VectorTileLayer.html#esri/layers/VectorTileLayer):

```js
new VectorTileLayer({
  url: "https://www.arcgis.com/sharing/rest/content/items/4cf7e1fb9f254dcda9c8fbadb15cf0f8/resources/styles/root.json"
});
```

And maybe:
```js
const layer = new VectorTileLayer({
 portalItem:{
   id: "1c365daf37a744fbad748b67aa69dac8"
 }
});
```